ecs-agent/netlib/data/network_client.go

```diff
@@ -94,3 +132,127 @@ func (ndc *networkDataClient) GetNetworkNamespace(netNSName string) (*tasknetwor
 
 	return netNS, nil
 }
+
+// AssignGeneveDstPort returns an unused destination port number for GENEVE interfaces.
+// By default for a particular VNI, it will return the default GENEVE destination port - 6081.
+// In case port 6081 is taken by another interface using the same VNI, it will chose a
+// random port from within the pre-configured range.
```

Comment:
What's the advantage of choosing a random port instead of a next available port?

Reply:
I have removed implementation of NetworkDataClient from the shared library because the agent is able to inject the same into the network builder.

To answer your question, maintaining a counter is takes more operations than choosing at random.